system/lib/libc/musl/src/thread/pthread_cancel.c

```diff
@@ -15,6 +14,7 @@ long __cancel()
 	return -ECANCELED;
 }
 
+#ifndef __EMSCRIPTEN__
```

Comment:
are the changes to this file related?

Reply:
It looks like this change is coming from the already landed https://github.com/emscripten-core/emscripten/commit/a1a755948a6e25c0fa62fc8fdcb89dc372618a63 commit. I think it's due to that this branch targets `remove_legacy_compiler_option` instead of the main branch.